crates/ruff/src/jupyter/notebook.rs

```diff
@@ -37,7 +27,15 @@ pub fn is_jupyter_notebook(path: &Path) -> bool {
         && cfg!(feature = "jupyter_notebook")
 }
 
-impl JupyterNotebook {
+#[derive(Debug)]
+pub struct Notebook {
+    pub index: JupyterIndex,
+    pub content: String,
+    raw: JupyterNotebook,
+    cell_offsets: Vec<TextSize>,
```

Comment:
would it make sense to move this to the index?

Reply:
The reason to keep the index separate is to have the ability to refresh _just_ the index for each iteration of auto-fix. The cell offsets cannot be updated just with the new cell content, we analyze each edit and update the offsets accordingly. This is done separately and is unrelated to the index.